packages/web-app-external/src/App.vue

```diff
@@ -0,0 +1,123 @@
+<template>
+  <main
+    class="uk-height-1-1"
+    :class="{
+      'uk-flex uk-flex-center uk-flex-middle': loading || loadingError
+    }"
+  >
+    <h1 class="oc-invisible-sr" v-text="pageTitle" />
+    <loading-screen v-if="loading" />
+    <error-screen v-else-if="loadingError" />
+    <iframe
+      v-if="appUrl && method === 'GET'"
+      :src="appUrl"
+      class="uk-width-1-1 uk-height-1-1"
+      :title="iFrameTitle"
+    />
+    <div v-if="appUrl && method === 'POST' && formParameters">
+      <form :action="appUrl" target="app-iframe" method="post">
+        <input ref="subm" type="submit" :value="formParameters" class="oc-hidden" />
+        <div v-for="(item, key, index) in formParameters" :key="index">
+          <input :name="key" :value="item" type="hidden" />
+        </div>
+      </form>
+      <iframe name="app-iframe" class="uk-width-1-1 uk-height-1-1" :title="iFrameTitle" />
+    </div>
+  </main>
+</template>
+
+<script>
+import { mapGetters } from 'vuex'
+import ErrorScreen from './components/ErrorScreen.vue'
+import LoadingScreen from './components/LoadingScreen.vue'
+
+export default {
+  name: 'ExternalApp',
+
+  components: {
+    ErrorScreen,
+    LoadingScreen
+  },
+
+  data: () => ({
+    loading: false,
+    loadingError: false,
+    appUrl: '',
+    method: '',
+    formParameters: {}
+  }),
+  computed: {
+    ...mapGetters(['getToken', 'configuration']),
+
+    pageTitle() {
+      const translated = this.$gettext('"%{appName}" app page')
+      return this.$gettextInterpolate(translated, {
+        appName: this.appName
+      })
+    },
+
+    iFrameTitle() {
+      const translated = this.$gettext('"%{appName}" app area')
+      return this.$gettextInterpolate(translated, {
+        appName: this.appName
+      })
+    },
+
+    appName() {
+      return this.$route.params.app
+    },
+    fileId() {
+      return this.$route.params.file_id + '=='
+    }
+  },
+  async created() {
+    this.loading = true
+
+    if (!this.getToken) {
+      this.loading = false
+      this.loadingError = true
+      return
+    }
+
+    const headers = new Headers()
+    headers.append('Authorization', 'Bearer ' + this.getToken)
+    headers.append('X-Requested-With', 'XMLHttpRequest')
+    const url =
+      this.configuration.server + 'app/open?file_id=' + this.fileId + '&app_name=' + this.appName
+
+    const response = await fetch(url, {
+      method: 'POST',
+      headers
+    })
+
+    if (response.status !== 200) {
```

Comment:
Is the return code really 200? Would have expected something else for a `POST`, but might still be possible/valid.